src/services/1inch.execute.tsx

```diff
@@ -15,28 +16,19 @@ export class OneInchExecutionManager {
     this.shouldContinue = val
   }
 
-  executeSwap = async (
-    signer: JsonRpcSigner,
-    action: Action,
-    estimate: Estimate,
-    srcAmount: BigNumber,
-    srcAddress: string,
-    destAddress: string,
-    updateStatus?: Function,
-    initialStatus?: Execution,
-    // eslint-disable-next-line max-params
-  ) => {
+  executeSwap = async ({ signer, step, updateStatus }: ExecuteSwapParams) => {
     // setup
+    const { action, execution } = step
     const fromChain = getChainById(action.fromChainId)
-    const { status, update } = initStatus(updateStatus, initialStatus)
+    const { status, update } = initStatus(updateStatus, execution)
     if (!this.shouldContinue) return status
     if (action.fromToken.id !== constants.AddressZero) {
       const contractAddress = await oneInch.getContractAddress()
       await checkAllowance(
         signer,
         fromChain,
         action.fromToken,
-        srcAmount.toString(),
+        action.fromAmount,
```

Comment:
the estimate contains an `approvalAddress` address you can use instead of `contractAddress` here.